devito/ir/clusters/algorithms.py

```diff
@@ -359,11 +359,10 @@ def pull_indexeds(expr, subs, mapper, parent=None):
         pull_indexeds(e, subs, {})
 
         # Construct temporaries and apply substitution to `e`, in cascade
-        pe = e
         for k, v in subs.items():
             processed.append(Eq(v, k))
-            pe = pe.xreplace({k: v})
-        processed.append(pe)
+            e = e.xreplace({k: v})
+        processed.append(e)
```

Comment:
notpicking: don't need processed anymore, just `cluster.rebuild(cluster.exprs)` not even sure if need the rebuild.